selfdrive/ui/qt/maps/map.cc

```diff
@@ -159,14 +159,14 @@ void MapWindow::updateState(const UIState &s) {
     }
   }
 
-  if (sm.updated("navRoute") && sm["navRoute"].getNavRoute().getCoordinates().size()) {
-    qWarning() << "Got new navRoute from navd. Opening map:" << allow_open;
-
-    // Only open the map on setting destination the first time
-    if (allow_open) {
-      emit requestSettings(false);
-      emit requestVisible(true); // Show map on destination set/change
-      allow_open = false;
+  if (sm.updated("navRoute")) {
+    auto nav_dest = coordinate_from_param("NavDestination");
+    if (std::exchange(current_nav_dest, nav_dest) != nav_dest) {
+      // Show map on destination set/change
+      if (current_nav_dest && !isVisible()) {
```

Comment:
can you combine to one check? just want to show if current_nav_dest and !nav_dest